pkg/leaderelection/leader_election.go

```diff
@@ -45,6 +43,10 @@ type Options struct {
 	// election resource will be created.
 	LeaderElectionNamespace string
 
+	// LeaderElectionUncachedNamespaceLookup if set to true will do an uncached read of the
+	// in-cluster namespace
+	LeaderElectionUncachedNamespaceLookup bool
```

Comment:
Hmm I'm not sure if this is worth surfacing as an option. It is kinda confusing as well (why cache this namespace in the first place? why do I care?). Is it really that hard to call `options.LeaderElectionNamespace = ReadInClusterNamespaceCached()` or `options.LeaderElectionNamespace = ReadInClusterNamespace()` depending on what mode you want when setting up `Options`, with a default to `ReadInClusterNamespace()`?

Reply:
Yeah, I agree. First of all was trying to fix the test, so added the uncached, but then after doing that, there's no reason to do the cached lookup for leader election where it's not going to be called very often unlike consumers within reconciliation loops. Deleting.